internal/config/config.go

```diff
@@ -0,0 +1,118 @@
+package config
+
+import (
+	"fmt"
+	"github.com/rs/zerolog"
+	"strings"
+	"time"
+
+	clowder "github.com/redhatinsights/app-common-go/pkg/api/v1"
+	"github.com/spf13/viper"
+)
+
+type DatabaseT struct {
+	Host        string
+	Port        uint16
+	Name        string
+	User        string
+	Password    string
+	MaxIdleTime time.Duration
+	MaxLifetime time.Duration
+	MaxOpenConn int
+	MaxIdleConn int
+	LogLevel    int
+}
+
+type LoggingT struct {
+	Level  int
+	Stdout bool
+}
+type CloudwatchT struct {
+	Enabled bool
+	Group   string
+	Stream  string
+}
+type FeatureFlagsT struct {
+	Environment string
+	ExitOnPanic bool
+}
+
+type GlobalConfigT struct {
+	Database   DatabaseT
+	Logging    LoggingT
+	Cloudwatch CloudwatchT
+	Features   FeatureFlagsT
+}
+
+var config GlobalConfigT
+var Database *DatabaseT
+var Logging *LoggingT
+var Cloudwatch *CloudwatchT
+var Features *FeatureFlagsT
+
+func init() {
+	var err error
+
+	viper.SetDefault("logging.level", 1)
+	viper.SetDefault("database.host", "localhost")
+	viper.SetDefault("database.port", 5432)
+	viper.SetDefault("database.loglevel", 1)
+	viper.SetDefault("cloudwatch.enabled", false)
+
+	if clowder.IsClowderEnabled() {
+		cfg := clowder.LoadedConfig
+
+		viper.Set("database.host", cfg.Database.Hostname)
+		viper.Set("database.port", cfg.Database.Port)
+		viper.Set("database.user", cfg.Database.Username)
+		viper.Set("database.password", cfg.Database.Password)
+		viper.Set("database.name", cfg.Database.Name)
+	} else {
+		viper.AddConfigPath("./configs")
+		viper.SetConfigName("defaults")
+		err = viper.ReadInConfig()
+		if err != nil {
+			panic(err)
+		}
+		viper.SetConfigName("local")
+		err = viper.MergeInConfig()
+		if err != nil {
+			fmt.Println("Could not read local.yaml", err)
+		}
+	}
+
+	viper.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
+	viper.AutomaticEnv()
+
+	err = viper.Unmarshal(&config)
+	if err != nil {
+		panic(err)
+	}
+
+	Database = &config.Database
+	Logging = &config.Logging
+	Cloudwatch = &config.Cloudwatch
+	Features = &config.Features
+}
+
+func IsDevelopment() bool {
+	return Features.Environment == "development"
+}
+
+func IsProduction() bool {
+	return Features.Environment == "production"
+}
+
+// DumpConfig writes configuration to a logger. It removes all secrets, however, it is never
+// recommended to call this function in production environments.
+func DumpConfig(logger zerolog.Logger) {
+	if IsProduction() {
+		logger.Warn().Msg("dumping configuration in production mode")
+	}
+	configCopy := config
+	configCopy.Database.Password = "*****"
+	// configCopy.AWS.Key = "*****"
+	// configCopy.AWS.Secret = "*****"
+	// configCopy.AWS.Session = "*****"
+	logger.Info().Msgf("%+v", configCopy)
```

Comment:
Oh for AWS (CloudWatch) we still use ENV variables, its okay for now. I will refactor that later.

Reply:
Yeah, if we could merge and follow-up that would be great, I'd love to unblock #7